tezos/new_context/src/working_tree/serializer.rs

```diff
@@ -204,6 +225,101 @@ pub fn serialize_entry(
     Ok(())
 }
 
+fn serialize_hash_id(hash_id: u32, output: &mut Vec<u8>) -> Result<usize, SerializationError> {
+    if hash_id & 0x7FFFFF == hash_id {
+        // The HashId fits in 23 bits
+
+        // Set `COMPACT_HASH_ID_BIT` so the deserializer knows the `HashId` is in 3 bytes
+        let hash_id: u32 = hash_id | COMPACT_HASH_ID_BIT;
+        let hash_id: [u8; 4] = hash_id.to_be_bytes();
+
+        output.write_all(&hash_id[1..])?;
+        Ok(3)
+    } else if hash_id & 0x7FFFFFFF == hash_id {
+        // HashId fits in 31 bits
+
+        output.write_all(&hash_id.to_be_bytes())?;
+        Ok(4)
+    } else {
+        // The HashId must not be 32 bits because we use the
+        // MSB to determine if the HashId is compact or not
+        Err(SerializationError::HashIdTooBig)
+    }
+}
+
+fn serialize_inode(
+    inode_id: InodeId,
+    output: &mut Vec<u8>,
+    hash_id: HashId,
+    storage: &Storage,
+    stats: &mut SerializeStats,
+    batch: &mut Vec<(HashId, Arc<[u8]>)>,
+    referenced_older_entries: &mut Vec<HashId>,
+) -> Result<(), SerializationError> {
+    output.clear();
+    let inode = storage.get_inode(inode_id).unwrap();
```

Comment:
There is an `unwrap()` here. If it is impossible for it to fail add a comment explaining so.

Reply:
Thanks, I missed that `unwrap`, the error is now handled correctly